### PR TITLE
Requirements & encoding

### PR DESCRIPTION
**Pull request by Nil-Andreu**
This is a small contribution on some things that I found interesting to take into consideration:

- **requirements.txt**: provide an easy way to create the environment.
- **.gitignore**: ignore the virtual environment as well as the data files
- **encoding**: specify the encoding with the I/O operations to make sure that the data is consistent across the Operating Systems

For me is totally fine if you want to review/discuss/reject them.
Let me know if it is needed anything else.

### CODE REVIEW

**Comment by karpathy** [comment]
Hi ty
- I like the encodings change
- I don't like boilerplate gitignores, I like when they're specific to repo
- pip installing pytorch would not grab the nightly I believe?

**Comment by Nil-Andreu** [comment]
> Hi ty
Hello!
> 
> * I like the encodings change
> * I don't like boilerplate gitignores, I like when they're specific to repo
Okay, could remove the things that are general and make it only specific to a repository.
> * pip installing pytorch would not grab the nightly I believe?
Not, but I saw the issue that this package was having with the 32 precision. So I could add the nightly as well in the requirements.

File: env.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+python3 -m venv venv
+venv/bin/pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cu117
```
Comment (acheong08) [inline env.sh:2]: This will fetch nightly version

**Review by acheong08 — CHANGES_REQUESTED** [review]

**Comment by bradezard131** [comment]
> * pip installing pytorch would not grab the nightly I believe?

You can point it at the nightly index URLs, but to do so via pip requires choosing a CUDA version to target as they have different URLs. This is a better dev experience for those who have that version, but worse for everyone else who doesn't.

**Comment by karpathy** [comment]
I'd like to have better requirements flow for the repo but don't love this specific approach. Will think on. Ty.

**Pull request closed without merging**